v2/pkg/genruntime/configmaps_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/format"
+)
+
+func Test_SerializeOptionalConfigMapReference(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		optionalRef OptionalConfigMapReference
+		expected    string
+	}{
+		{
+			name: "Serialized as string",
+			optionalRef: OptionalConfigMapReference{
+				Value: to.StringPtr("test"),
+			},
+			expected: `"test"`,
+		},
+		{
+			name: "Serialized as ConfigMapReference",
+			optionalRef: OptionalConfigMapReference{
+				Ref: &ConfigMapReference{
+					Name: "foo",
+					Key:  "key",
+				},
+			},
+			expected: `{"name":"foo","key":"key"}`,
+		},
+		{
+			name:        "Serialized when empty", // This isn't really an expected scenario, but we try to do something sensible anyway.
+			optionalRef: OptionalConfigMapReference{},
+			expected:    `{}`,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			g := NewGomegaWithT(t)
+
+			// Serialize as non-ptr
+			result, err := json.Marshal(tt.optionalRef)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(string(result)).To(Equal(tt.expected))
+
+			// Also serialize as ptr, they should be the same
+			result, err = json.Marshal(&tt.optionalRef)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(string(result)).To(Equal(tt.expected))
+		})
+	}
+}
+
+func Test_DeserializeOptionalConfigMapReference(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
```

Comment:
Elsewhere we've called this `cases` and used `c` as the local variable.

Reply:
Fixed